pbs/pbsrequest.go

```diff
@@ -328,3 +332,72 @@ func (p PBSRequest) String() string {
 	b, _ := json.MarshalIndent(p, "", "    ")
 	return string(b)
 }
+
+func CompareVersions(v1, v2 string) int {
+	var x, r, l int = 0, 0, 0
+	v1Digits := strings.Split(v1, ".")
+	v2Digits := strings.Split(v2, ".")
+	len1, len2 := len(v1Digits), len(v2Digits)
+
+	if len1 > len2 {
+		x = len1
+	} else {
+		x = len2
+	}
+
+	for i := 0; i < x; i++ {
+		if i < len1 && i < len2 {
+			if v1Digits[i] == v2Digits[i] {
+				continue
+			}
+		}
+
+		r = 0
+		if i < len1 {
+			r = NumVersion(v1Digits[i])
+		}
+
+		l = 0
+		if i < len2 {
+			l = NumVersion(v2Digits[i])
+		}
+
+		if r < l {
+			return -1
+		} else if r > l {
+			return 1
+		}
+	}
+
+	return 0
+
+}
+
+var specialForms = map[string]int{
```

Comment:
I'm a little confused by these... is there some reason we can't stick to standard semantic versioning?

If we do, there are a bunch of libraries in Go which already do this for you. For example:

https://github.com/blang/semver
https://github.com/hashicorp/go-version

Reply:
Will check the libraries.